ec/ecresource/deploymentresource/elasticsearch/v2/schema.go

```diff
@@ -255,6 +255,14 @@ func elasticsearchTopologyAutoscalingSchema(topologyAttributeName string) schema
 					UseTopologyStateForUnknown(topologyAttributeName),
 				},
 			},
+			"autoscale": schema.BoolAttribute{
```

Comment:
Should we enable the attribute for `ml` only?

Reply:
So as for today, `ml` is the only tier which can be autoscaled exclusively, but from the API perspective, this parameter is valid for all the topology elements. It's just that ESS is going to eventually reject it:

```
│ api error: 1 error occurred:
│ 	* deployments.invalid_autoscaling_config: Invalid autoscaling configuration in deployment: [The `autoscaling_tier_override` field can be used only for ML topology
│ elements, but found in [hot_content]]
```
Which just ends up with a meaningful error message and doesn't harm the deployment.

Additionally, if we ever decide in the future to enable this for the remaining tiers (or at even a subset of these), we'll end up updating the provider accordingly. From what I've chatted with @Kushmaro, our preference is to avoid client-side logic in the provider, and I feel such a behaviour would be an instance of it. 

But there is indeed a value in throwing the error at the declaration level, even before sending that request to the ESS API ! 

Honestly I don't have that much of a preference here, so I'll defer to your decision 🙂

Reply:
If Omer is fine with that, I'm fine too 🙂.

Reply:
I'm good with this, since the server-side instructs the user on what's wrong :)